runtime/lua/vim/lsp/buf.lua

```diff
@@ -104,6 +141,13 @@ function M.formatting_sync(options, timeout_ms)
   vim.lsp.util.apply_text_edits(result)
 end
 
+--- Perform |vim.lsp.buf.formatting()| synchronously.
+---
+--- Useful for running on save, to make sure buffer is formatted prior to being
+--- saved. {timeout_ms} is passed on to |vim.lsp.buf_request_sync()|.
```

Comment:
Maybe we could just add an example with the `<pre>` style w/ an autocommand to show how to do it?

Reply:
Sure, will do.